test/note_display.test.js

```diff
@@ -0,0 +1,530 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-undef */
+import * as Factory from './lib/factory.js';
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+describe('note display criteria', function () {
+  before(async function () {
+    const crypto = new SNWebCrypto();
+    Uuid.SetGenerators(
+      crypto.generateUUIDSync,
+      crypto.generateUUID
+    );
+  });
+
+  beforeEach(function () {
+    this.modelManager = new PayloadManager();
+    this.itemManager = new ItemManager(this.modelManager);
+
+    this.createNote = async () => {
+      return this.itemManager.createItem(
+        ContentType.Note,
+        {
+          title: 'hello',
+          text: 'world'
+        }
+      );
+    };
+
+    this.createTag = async (notes = [], title = 'thoughts') => {
+      const references = notes.map((note) => {
+        return {
+          uuid: note.uuid,
+          content_type: note.content_type
+        };
+      });
+      return this.itemManager.createItem(
+        ContentType.Tag,
+        {
+          title: title,
+          references: references
+        }
+      );
+    };
+  });
+
+  const emptyCollection = function () {
+    const collection = new ItemCollection();
+    collection.setDisplayOptions(
+      ContentType.Note,
+      CollectionSort.CreatedAt,
+      'asc'
+    );
+    return collection;
+  }
+
+  const collectionWithNotes = function (titles = [], bodies = []) {
+    const collection = emptyCollection();
+    const notes = [];
+    titles.forEach((title, index) => {
+      notes.push(CreateItemFromPayload(Factory.createNotePayload(title, bodies[index])));
+    })
+    collection.set(notes);
+    return collection;
+  }
+
+  it('display criteria copying', async function () {
```

Comment:
Hmm, I feel like a lot of the tests names in this file could describe their expect results/outputs in more detail. 

Taking this first one as an example, if you read it with the `it` at the beginning you get this: _It display criteria copying._
It does not explain the exact behavior it's testing for. In this specific case I think something like:
`it('creates a criteria copy with an updated parameter')` explains what's actually happening (and not happening) in the test better, and opens the door to think about other things to test (like making sure that other properties in the criteria copy have not been altered)

What do you think?

Reply:
Yeah, I thought about that, but essentially it's like the utils.test.js approach, where each test is the function name. The `it` is too restrictive imo. What I think makes more sense to envision is the name of a test followed by a green checkmark. So `note display criteria.string query text ✅` tells me a good amount.